src/core/Table/Table.tsx

```diff
@@ -337,7 +343,13 @@ export const Table = <
       ref={(element) => setOwnerDocument(element?.ownerDocument)}
       id={id}
       {...getTableProps({
-        className: cx('iui-table', className),
+        className: cx(
+          'iui-table',
+          {
+            [`iui-${density}`]: density !== 'default',
+          },
```

Comment:
This could be in one line (unless you prefer it this way)
```suggestion
          { [`iui-${density}`]: density !== 'default' },
```